--- a/cloudinit/config/cc_ssh_vyatta.py
+++ b/cloudinit/config/cc_ssh_vyatta.py
@@ -76,6 +76,10 @@
         except:
             util.logexc(log, "Failed to generate SSH keys");
 
+    # Restart SSH after creating/placing host keys
+    cmd = ['systemctl', 'restart', 'ssh.service']
+    util.subp(cmd, capture=False)
+
     try:
         (users, _groups) = ug_util.normalize_users_groups(cfg, cloud.distro)
         (user, _user_config) = ug_util.extract_default(users)
