Run config and final stages with lu

---

--- a/systemd/cloud-config.service.tmpl
+++ b/systemd/cloud-config.service.tmpl
@@ -8,7 +8,7 @@
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/sg vyattacfg "/usr/bin/cloud-init modules --mode=config"
+ExecStart=/opt/vyatta/sbin/lu -user configd -- /usr/bin/cloud-init modules --mode=config
 RemainAfterExit=yes
 TimeoutSec=0
 
--- a/systemd/cloud-final.service.tmpl
+++ b/systemd/cloud-final.service.tmpl
@@ -12,7 +12,7 @@
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/sg vyattacfg "/usr/bin/cloud-init modules --mode=final"
+ExecStart=/opt/vyatta/sbin/lu -user configd -- /usr/bin/cloud-init modules --mode=final
 RemainAfterExit=yes
 TimeoutSec=0
 KillMode=process
