--- a/cloudinit/util.py
+++ b/cloudinit/util.py
@@ -832,7 +832,7 @@
         if os.path.isfile(exe_path) and os.access(exe_path, os.X_OK):
             attempted.append(exe_path)
             try:
-                subp(prefix + [exe_path], capture=False)
+                subp(['/bin/vcli', '-f', exe_path], capture=False)
             except ProcessExecutionError as e:
                 logexc(LOG, "Failed running %s [%s]", exe_path, e.exit_code)
                 failed.append(e)
@@ -2054,7 +2054,7 @@
 
     content = ''
     if add_header:
-        content += "#!/bin/sh\n"
+        content += "#!/bin/vcli -f\n"
     escaped = "%s%s%s%s" % ("'", '\\', "'", "'")
     cmds_made = 0
     for args in cmdlist:
