--- a/cloudinit/util.py
+++ b/cloudinit/util.py
@@ -832,7 +832,7 @@
         if os.path.isfile(exe_path) and os.access(exe_path, os.X_OK):
             attempted.append(exe_path)
             try:
-                subp(prefix + [exe_path], capture=False)
+                subp(['/bin/vcli', '-f', exe_path], capture=False)
             except ProcessExecutionError as e:
                 logexc(LOG, "Failed running %s [%s]", exe_path, e.exit_code)
                 failed.append(e)
@@ -2054,7 +2054,7 @@
 
     content = ''
     if add_header:
-        content += "#!/bin/sh\n"
+        content += "#!/bin/vcli -f\n"
     escaped = "%s%s%s%s" % ("'", '\\', "'", "'")
     cmds_made = 0
     for args in cmdlist:
--- a/tests/unittests/test_handler/test_handler_runcmd.py
+++ b/tests/unittests/test_handler/test_handler_runcmd.py
@@ -95,7 +95,7 @@
         runcmd_file = os.path.join(
             self.new_root,
             'var/lib/cloud/instances/iid-datasource-none/scripts/runcmd')
-        self.assertEqual("#!/bin/sh\n'ls' '/'\n", util.load_file(runcmd_file))
+        self.assertEqual("#!/bin/vcli -f\n'ls' '/'\n", util.load_file(runcmd_file))
         file_stat = os.stat(runcmd_file)
         self.assertEqual(0o700, stat.S_IMODE(file_stat.st_mode))
 
