--- a/systemd/cloud-config.service
+++ b/systemd/cloud-config.service
@@ -1,7 +1,8 @@
 [Unit]
 Description=Apply the settings specified in cloud-config
-After=network-online.target cloud-config.target syslog.target
+After=network-online.target cloud-config.target syslog.target configd.service
 Wants=network-online.target cloud-config.target
+ConditionKernelCommandLine=cloud-init
 
 [Service]
 Type=oneshot
--- a/systemd/cloud-final.service
+++ b/systemd/cloud-final.service
@@ -1,7 +1,9 @@
 [Unit]
 Description=Execute cloud user/final scripts
 After=network-online.target cloud-config.service syslog.target rc-local.service
+Requires=config-loaded.target
 Wants=network-online.target cloud-config.service
+ConditionKernelCommandLine=cloud-init
 
 [Service]
 Type=oneshot
--- a/systemd/cloud-init-local.service
+++ b/systemd/cloud-init-local.service
@@ -2,6 +2,7 @@
 Description=Initial cloud-init job (pre-networking)
 Wants=local-fs.target
 After=local-fs.target
+ConditionKernelCommandLine=cloud-init
 
 [Service]
 Type=oneshot
--- a/systemd/cloud-init.service
+++ b/systemd/cloud-init.service
@@ -1,9 +1,10 @@
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
 After=local-fs.target network-online.target cloud-init-local.service
-Before=sshd.service sshd-keygen.service systemd-user-sessions.service
+Before=sshd.service sshd-keygen.service systemd-user-sessions.service configd.service
 Requires=network-online.target
 Wants=local-fs.target cloud-init-local.service sshd.service sshd-keygen.service
+ConditionKernelCommandLine=cloud-init
 
 [Service]
 Type=oneshot
